receiver/azuremonitorreceiver/scraper.go

```diff
@@ -134,27 +134,27 @@ func (s *azureScraper) getArmClientOptions() *arm.ClientOptions {
 	return &options
 }
 
-func (s *azureScraper) getArmClient() ArmClient {
+func (s *azureScraper) getArmClient() armClient {
```

Comment:
while it's ok to keep `armClient` and  `metricsDefinitionsClientInterface` interfaces to help mock objects in tests it's more idiomatic to return concrete types. 
also not related to this change, but this func is used only in `start` which has proper error handling (returning error) but we are silently failing here. this is just delaying problem to few seconds later risking panics when client is nil

Reply:
Appreciate you raising these points @michalpristas as this improves correctness and maintainability. To keep this PR focused I'd like to open issues based on your points and assign myself to fix in separate PRs.